modules/javafx.graphics/src/test/java/test/com/sun/javafx/pgstub/StubFontLoader.java

```diff
@@ -47,123 +47,127 @@ public class StubFontLoader extends FontLoader {
-            FontHelper.setNativeFont(font, nativeFont, font.getName(), "Amble Cn",
-                    "Bold Italic");
-        } else if (name.equals("amble light")) {
-            FontHelper.setNativeFont(font, nativeFont, font.getName(), "Amble Lt", "Regular");
-        } else if (name.equals("amble light italic")) {
-            FontHelper.setNativeFont(font, nativeFont, font.getName(), "Amble Lt", "Italic");
-        } else if (name.equals("amble light condensed")) {
-            FontHelper.setNativeFont(font, nativeFont, font.getName(), "Amble LtCn",
-                    "Regular");
-        } else if (name.equals("amble light condensed italic")) {
-            FontHelper.setNativeFont(font, nativeFont, font.getName(), "Amble LtCn",
-                    "Italic");
+        switch (name) {
+            case "system regular" -> FontHelper.setNativeFont(font, nativeFont, font.getName(), "System", "Regular");
+            case "system bold" -> FontHelper.setNativeFont(font, nativeFont, font.getName(), "System", "Bold");
+            case "system italic" -> FontHelper.setNativeFont(font, nativeFont, font.getName(), "System", "Italic");
+            case "system bold italic" ->
+                    FontHelper.setNativeFont(font, nativeFont, font.getName(), "System", "Bold Italic");
+            case "amble regular" -> FontHelper.setNativeFont(font, nativeFont, font.getName(), "Amble", "Regular");
+            case "amble bold" -> FontHelper.setNativeFont(font, nativeFont, font.getName(), "Amble", "Bold");
+            case "amble italic" -> FontHelper.setNativeFont(font, nativeFont, font.getName(), "Amble", "Italic");
+            case "amble bold italic" ->
+                    FontHelper.setNativeFont(font, nativeFont, font.getName(), "Amble", "Bold Italic");
+            case "amble condensed" -> FontHelper.setNativeFont(font, nativeFont, font.getName(), "Amble Cn", "Regular");
+            case "amble bold condensed" ->
+                    FontHelper.setNativeFont(font, nativeFont, font.getName(), "Amble Cn", "Bold");
+            case "amble condensed italic" ->
+                    FontHelper.setNativeFont(font, nativeFont, font.getName(), "Amble Cn", "Italic");
+            case "amble bold condensed italic" ->
+                    FontHelper.setNativeFont(font, nativeFont, font.getName(), "Amble Cn", "Bold Italic");
+            case "amble light" -> FontHelper.setNativeFont(font, nativeFont, font.getName(), "Amble Lt", "Regular");
+            case "amble light italic" ->
+                    FontHelper.setNativeFont(font, nativeFont, font.getName(), "Amble Lt", "Italic");
+            case "amble light condensed" ->
+                    FontHelper.setNativeFont(font, nativeFont, font.getName(), "Amble LtCn", "Regular");
+            case "amble light condensed italic" ->
+                    FontHelper.setNativeFont(font, nativeFont, font.getName(), "Amble LtCn", "Italic");
```

Comment:
does it make sense to handle/log the default case, or this is indeed the desired code (i.e. a no-op)?

Reply:
Right now, if we try to load a font that is not known by the test setup, the font will not have a 'native font' since no branch will get executed. With that, the width calculation will return 0.

Just checked the Prism font code, there the code will fallback to the system font if it can not load the specified font (family). 
That would also be an idea here, not sure if it will have consequences on some tests. Also okay for me. What do you think?

Reply:
I gave the fallback font a shot, since this is closer to the `PrismFontLoader`, which is good when we try to be as close a possible in the tests.